[frameit] Add "Screenshot orientation" section to `frameit` documentation (#14164)
fastlane/fastlane: fastlane/lib/fastlane/actions/docs/frame_screenshots.md

- These `.strings` files **MUST** be utf-8 (UTF-8) or utf-16 encoded (UTF-16 BE with BOM). They also must begin with an empty line. If you are having trouble see [issue #1740](https://github.com/fastlane/fastlane/issues/1740)
- You **MUST** provide a background if you want titles. _frameit_ will not add the tiles if a background is not specified.

### Screenshot orientation

> By default _frameit_ adds a frame to your screenshot based on an orientation you took it. For a portrait (vertical orientation) it is going to add portrait frame and for a landscape (horizontal orientation) - landscape left.

Reviewer: What does "landscape left" actually mean here?

Author: According to Apple's [documentation](https://developer.apple.com/documentation/uikit/uiinterfaceorientation/landscapeleft):

> The device is in landscape mode, with the device held upright and the Home button on the left side.

So maybe it's worth to add this sentence? 

```
... for a landscape (horizontal orientation) - landscape left. The device is in landscape mode, with the device held upright and the Home button on the left side.
```

Not sure if it's okay to use the same sentence from Apple's documentations without including the source, so maybe better you can insert a link to the docs: `... - [landscape left](https://developer.apple.com/documentation/uikit/uiinterfaceorientation/landscapeleft)` ? 🤔

Reviewer: Simple:
```suggestion
By default _frameit_ adds a frame to your screenshot based on an orientation you took it. For a portrait (vertical orientation) it is going to add portrait frame and for a landscape (horizontal orientation) - landscape left (= [Home button on the left side](https://developer.apple.com/documentation/uikit/uiinterfaceorientation/landscapeleft)).
```

> If you'd like to get a frame in a landscape right for your horizontal screenshots, you need to add `force_landscaperight` at the end of your screenshot's filename. 

Reviewer: Maybe start this paragraph with "One way to override the default behavior ..." and then describe that one can edit the file name in this specific way to get that specific result.

### `force_orientation_block`

> In a case you don't want to rename your screenshots or default behaviour does not meet your requirements, there's a way to customise it. In order to do so, you need to set `force_orientation_block` parameter.

Reviewer: ```suggestion
If the default behavior doesn't fit your needs and you don't want or can't rename your screenshots, you can customize _frameit_'s orientation behavior by setting a `force_orientation_block` parameter.
```

Reviewer: You can accept this change here in the GitHub UI and it will automatically be added. (Caution: You have to fetch if you have a local checkout before committing again, otherwise you might get a merge conflict)

> In a case you don't want to rename your screenshots or default behaviour does not meet your requirements, there's a way to customise it. In order to do so, you need to set `force_orientation_block` parameter.

Reviewer: Maybe add some additional information how this block works (code executed when the action is run) and what possible return values there are, and what the effect of each are.

### Examples

> ```ruby

Reviewer: Some comment on what this does in plain English maybe?

Author: Hmm, not sure if an additional comment will add more value. 

```
# It frames the screenshots in landscape right whenever the filename contains `landscape` word
...
```

Code is pretty simple anyway so I'd not add an extra noise 🤷‍♂️ WDYT?

Reviewer: Yep, that sounds good. Keep in mind that typical fastlane and frameit users usually do not know ruby! They just use the "syntax" of the `Fastfile` they see somewhere which makes intuitive sense to all developers. The blocks... not so much. (e.g. missing `return` etc)

Author: That makes sense 🙇 Actually I was pretty sure most of the fastlane users know ruby 😅

Reviewer: **I** don't really know ruby 🙊

Reviewer 2: ^ Lies lol

frameit(
  path: "./fastlane/screenshots",
  force_orientation_block: proc do |filename|
    case filename

>       when "1Login" 

Reviewer 2: This may be nitpicky but we may want to rework this example as I don't think it will actually work if copy pasted 😇 `filename` returns something like "iPad Pro (12.9-inch)-01LoginScreen" so this example may be confusing if copy pasted 🤔

Author: Actually it depends on the way the users generate the screenshots. The way I was testing the parameter was just running a sample application, hit `cmd + s` and move the screenshot to the proper catalog. Then I renamed it as I wanted. However I can imagine that most common scenario to generate screenshots is `snapshot` tool. In that case, indeed would be great provide better context to our example 🙂 

```ruby
frameit(
  path: "./fastlane/screenshots",
  force_orientation_block: proc do |filename|
    case filename
      when "iPad Pro (12.9-inch)-01LoginScreen" 
        :landscape_right
      when "iPhone 6 Plus-01LoginScreen"
        : portrait
      # and so on
    end
  end
)
```
WDYT @joshdholtz ? 🤔

Reviewer: Better. Maybe we can also add an example that matches only part of the filename to decide?

Reviewer: Oh right - ignore that and just keep these two examples.

        :landscape_left
      when "2Profile"
        :landscape_right
      # and so on
    end
  end
)
```

> ```ruby

Reviewer: Some comment on what this does in plain English maybe?

Reviewer: You switched the both comments, but at the correct example those look good :)

>         : portrait

Reviewer: ```suggestion
        :portrait
```
